expression/constant.go

```diff
@@ -88,6 +88,253 @@ func (c *Constant) GetType() *types.FieldType {
 	return c.RetType
 }
 
+// Vectorized returns if this expression supports vectorized evaluation.
+func (c *Constant) Vectorized() bool { return true }
+
+// VecEval evaluates this expression in a vectorized manner.
+func (c *Constant) VecEval(ctx sessionctx.Context, chk *chunk.Chunk, result *chunk.Column) error {
+	if c.DeferredExpr == nil {
+		return genVecFromConstExpr(ctx, c, chk, result)
+	}
+	return c.vecEvalWithDeffer(ctx, chk, result)
+}
+
+func (c *Constant) vecEvalWithDeffer(ctx sessionctx.Context, chk *chunk.Chunk, result *chunk.Column) (err error) {
+	result.Reset()
+	n := resultLength(chk)
+	sel := chk.Sel()
+
+	switch c.GetType().EvalType() {
+	case types.ETInt:
+		if sel == nil {
+			for i := 0; i < n; i++ {
+				v, isNull, err := c.EvalInt(ctx, chunk.Row{})
```

Comment:
can we move this line out of the for loop?

Reply:
We can't do that since some scalar functions like `SYSDATE`, `UUID` should produce different values for each row.

Reply:
Those expressions are not constant.

Reply:
Maybe the `vecEvalWithDeffer` is not needed.

Reply:
But there are some cases which fold these scalar functions as `Constant` and put them into `Defer`, otherwise why there is `Defer`?

Reply:
`Defer` is used for plan cache for prepared statements.

Reply:
I checked it out again, non-deterministic functions are rewritten to constant with DeferExpression for plan cache.

Reply:
Fixed.